### PR TITLE
ci: add kafka integration test

### PR DESCRIPTION
**Pull request by leoppro**
### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change
 - Has persistent data change

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation
 - Need to update the `tidb-cdc/cdc-ansible`
 - Need to be included in the release note

### CODE REVIEW

File: scripts/jenkins_ci/integration_test_common.groovy [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+        prepares["download third binaries"] = {
+            node ("${GO_TEST_SLAVE}") {
+                container("golang") {
+                    def ws = pwd()
+                    deleteDir()
+
+                    sh """
+                        mkdir -p third_bin
+                        mkdir -p tmp
+                        curl ${FILE_SERVER_URL}/download/builds/pingcap/tidb/700d9def026185fe836dd56b0c39e0b4df3c320b/centos7/tidb-server.tar.gz | tar xz -C ./tmp bin/tidb-server
```
Comment (amyangfei) [inline scripts/jenkins_ci/integration_test_common.groovy]: maybe we can refine the binary used here. (we can refine it in another PR)
- fetch `tidb_sha1` from `${FILE_SERVER_URL}/download/refs/pingcap/tidb/${TIDB_BRANCH}/sha1")`
- and use any commit hash by `branch` or `release tag`

File: tests/cdc/run.sh [hunk 2]
```diff
@@ -19,7 +20,17 @@ function prepare() {
...
 
     run_cdc_server $WORK_DIR $CDC_BINARY
-    cdc cli changefeed create --start-ts=$start_ts --sink-uri="dsn://root@tcp(127.0.0.1:3306)/"
+
+    TOPIC_NAME="ticdc-cdc-test-$RANDOM"
+    case $SINK_TYPE in
+        mysql) SINK_URI="mysql://root@127.0.0.1:3306/";;
+        kafka) SINK_URI="kafka://127.0.0.1:9092/$TOPIC_NAME?partition-num=4";;
+        *) SINK_URI="mysql://root@127.0.0.1:3306/";;
+    esac
```
Comment (amyangfei) [inline tests/cdc/run.sh]:
```suggestion
    case $SINK_TYPE in
        kafka) SINK_URI="kafka://127.0.0.1:9092/$TOPIC_NAME?partition-num=4";;
        mysql) ;&  
        *) SINK_URI="mysql://root@127.0.0.1:3306/";;
    esac
```
we can save one line with the `fall-through` trick

File: scripts/jenkins_ci/integration_test_common.groovy (hunk 1 above)
Comment (leoppro) [reply scripts/jenkins_ci/integration_test_common.groovy]: fine i will refine it in another PR

File: tests/cdc/run.sh (hunk 2 above)
Comment (leoppro) [reply tests/cdc/run.sh]: ok I'm not familiar with bash syntax

**Pull request merged**